src/main/java/pages/yandex/MailPage.java

```diff
@@ -1,27 +1,46 @@
 package pages.yandex;
 
-import lombok.Getter;
-import lombok.Setter;
+import helperClasses.SingletonBrowser;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
 
-@Getter
-@Setter
 public class MailPage {
-    private static  final By YANDEX_LOGO = By.cssSelector(".PSHeaderLogo360_theme_light");
-    private static final By WRITE_EMAIL = By.xpath("//i[contains(text(),'Compose')]");
+    private static final By WRITE_EMAIL = By.xpath("//a[contains(@href,'compose')]");
     private static final By LIGHT_VERSION_BUTTON = By.cssSelector("span.mail-App-Footer-Item.mail-App-Footer-Item_lite");
-    private final By CUSTOMER_NAME = By.cssSelector(".b-pseudo-link");
+    private static final By CUSTOMER_NAME = By.cssSelector(".b-pseudo-link");
+    private static final By LOG_OUT_LINK = By.cssSelector("a.b-header__link_exit");
+    private static final By USER_NAME = By.xpath("//div[@class='b-head-user']//a[contains(@href,'profile')]");
+
+    private final WebDriver driver;
+
+    public MailPage() {
+        this.driver = SingletonBrowser.getInstance().getDriver();
+    }
 
-    final WebDriver driver;
+    public By getUSER_NAME() {
+        return USER_NAME;
+    }
 
-    public MailPage(WebDriver driver) {
-        this.driver = driver;
+    public By getCUSTOMER_NAME() {
+        return CUSTOMER_NAME;
     }
```

Comment:
Your page object is incorrect, if class has public methods which return By/WebElement. Update